### PR TITLE
[WIP] [v6r20?] Replacing pyGSI with M2Crypto

### PR DESCRIPTION
**Pull request by miloszz**
For now: handling certificates (no X509Request yet!)

- M2Crypto for everything, except
- asn1 for decoding VOMS extension

(I hope code isn't contaminated too much with v6r18, there could be some leftovers after rebase and recovering from backup)

### CODE REVIEW

File: Core/DISET/private/BaseClient.py [hunk 1]
```diff
@@ -574,17 +574,17 @@ def __setKeepAliveLapse( self ):
...
+        newKwargs is an updated copy of kwargs:
+          * kwargs has already been updated by all the initialization functions
+          * if a delegated DN is not set, but is found in __threadConfig or in the info returned by the protocol
+            sanitiy check, then we replace it (KW_DELEGATED_DN)
+          * Same goes for the delegated group KW_DELEGATED_GROUP. In the case of a host,
+            the value is the same as for VAL_EXTRA_CREDENTIALS_HOST
+          * if set, we remove the useCertificates (KW_USE_CERTIFICATES) in newKwargs (not sure to know why)
 
-	This method is just used to return information in case of error in the InnerRPCClient
+        This method is just used to return information in case of error in the InnerRPCClient
```
Comment (chaen) [inline Core/DISET/private/BaseClient.py]: all this is just formating. Do you autoformat ?

File: Core/Security/X509Certificate.py [hunk 2]
```diff
@@ -2,20 +2,50 @@
...
+import M2Crypto
+import asn1
+import datetime
+
 import os
 from DIRAC import S_OK, S_ERROR
 from DIRAC.Core.Utilities import Time
 from DIRAC.Core.Utilities import DErrno
 from DIRAC.ConfigurationSystem.Client.Helpers import Registry
+from DIRAC import gLogger
```
Comment (chaen) [inline Core/Security/X509Certificate.py]: Unused import ?

File: Core/Security/X509Certificate.py [hunk 3]
```diff
@@ -32,9 +62,8 @@ def loadFromFile( self, certLocation ):
     Return : S_OK / S_ERROR
     """
     try:
-      fd = file( certLocation )
-      pemData = fd.read()
-      fd.close()
+      with file( certLocation ) as fd:
```
Comment (chaen) [inline Core/Security/X509Certificate.py]: What is this form ?
Isn't it normally
```with open(certLocaltion, 'r') as fd:
   #do something
```
?

File: Core/DISET/private/BaseClient.py (hunk 1 above)
Comment (miloszz) [inline Core/DISET/private/BaseClient.py]: Honestly, I don't know. I recovered this file from backup and copied over integration one.

File: Core/Security/X509Certificate.py [hunk 4]
```diff
@@ -92,7 +123,7 @@ def getSubjectDN( self ):
     """
     if not self.__valid:
       return S_ERROR( DErrno.ENOCERT )
-    return S_OK( self.__certObj.get_subject().one_line() )
+    return S_OK( str( self.__certObj.get_subject() ) )
```
Comment (chaen) [inline Core/Security/X509Certificate.py]: why do you have to do all these str cast / what is the output type normally ?

File: Core/Security/X509Chain.py [hunk 5]
```diff
@@ -9,17 +9,27 @@
...
+import time
+import GSI # XXX Still needed for some parts I haven't finished yet
 
 from DIRAC import S_OK, S_ERROR
 from DIRAC.Core.Utilities import DErrno
-from DIRAC.Core.Security.X509Certificate import X509Certificate
 from DIRAC.ConfigurationSystem.Client.Helpers import Registry
+from DIRAC.Core.Security.X509Certificate import X509Certificate
+
+#from xext import xext
```
Comment (chaen) [inline Core/Security/X509Chain.py]: remove comment

File: Core/Security/X509Chain.py [hunk 6]
```diff
@@ -9,17 +9,27 @@
...
+#print xext("1.2.42.42", "diracGroup", "DIRAC group")
 
 random.seed()
 
 class X509Chain( object ):
 
+  __pass = None
+  def __getPass(self):
```
Comment (chaen) [inline Core/Security/X509Chain.py]: what is the use of this I do not get it

File: Core/Security/X509Chain.py [hunk 7]
```diff
@@ -81,6 +96,13 @@ def loadChainFromString( self, data, dataFormat = crypto.FILETYPE_PEM ):
     self.__checkProxyness()
     return S_OK()
 
+  def __listFromString( self, certString, format = M2Crypto.X509.FORMAT_PEM ):
```
Comment (chaen) [inline Core/Security/X509Chain.py]: you may want a better name for that one :)

File: Core/Security/X509Chain.py (hunk 6 above)
Comment (miloszz) [inline Core/Security/X509Chain.py]: I'm pretty sure there was a reason. But I don't see it now.

File: Core/Security/X509Certificate.py (hunk 4 above)
Comment (miloszz) [inline Core/Security/X509Certificate.py]: It's a name object. And I wanted to keep the old interface, where getSubjectDN returned string.

**Comment by fstagni** [comment]
203 files changed and way more commits... IMHO a new PR from a new branch where you cherry-pick your changed on top of integration is the easiest/safest option.

**Comment by miloszz** [comment]
Yup, I will make a cleanup, just wanted to make sure that I didn't break anything.

**Pull request closed without merging**